src-input/duk_bi_encoding.c

```diff
@@ -242,39 +283,35 @@ DUK_INTERNAL duk_ret_t duk_bi_textdecoder_constructor(duk_context *ctx) {
 	return 0;
 }
 
-DUK_INTERNAL duk_ret_t duk_bi_textdecoder_prototype_encoding_getter(duk_context *ctx) {
+/* Get TextDecoder context from 'this'; leaves garbage on stack. */
+DUK_LOCAL duk__decode_context *duk__get_textdecoder_context(duk_context *ctx) {
 	duk__decode_context *dec_ctx;
-
 	duk_push_this(ctx);
 	duk_get_prop_string(ctx, -1, "\xff" "Context");
 	dec_ctx = (duk__decode_context *) duk_require_buffer(ctx, -1, NULL);
 	DUK_ASSERT(dec_ctx != NULL);
+	return dec_ctx;
+}
 
-	/* XXX: change to look up from dec_ctx once other encodings are supported */
-	DUK_UNREF(dec_ctx);
+DUK_INTERNAL duk_ret_t duk_bi_textdecoder_prototype_encoding_getter(duk_context *ctx) {
+	/* Encoding is now fixed, so no need to look up _Context. */
```

Comment:
I originally left that fake `dec_ctx` lookup in just in case someone calls the getter directly, I still wanted it to TypeError if `this` was something other than a TextDecoder instance.  I guess that's not very footprint friendly though, and admittedly a bit of a contrived use case.